contracts/crypto-verify/src/contract.rs

```diff
@@ -59,6 +69,20 @@ pub fn query_verify(
     }
 }
 
+pub fn query_verify_tendermint(
+    deps: Deps,
+    message: &[u8],
+    signature: &[u8],
+    public_key: &[u8],
+) -> StdResult<VerifyResponse> {
+    // Verification
+    let result = deps.api.ed25519_verify(message, signature, public_key);
```

Comment:
You can do error "throwing" like this when the error types implement `From<Source> for Target`:

```suggestion
    let verifies = deps.api.ed25519_verify(message, signature, public_key)?;
```